src/components/Topic.jsx

```diff
@@ -0,0 +1,57 @@
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+import axios from "axios";
+
+function Topic() {
+  const { single_topic } = useParams();
+  const [topicList, setTopicList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(true);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  useEffect(() => {
+    axios
+      .get("https://nc-news-pa.herokuapp.com/api/articles")
+      .then((res) => {
+        const filteredTopic = res.data.articles.filter(
+          (article) => article.topic === single_topic
+        );
+        setTopicList(filteredTopic);
+        setIsLoading(false);
+        setError(false);
+      })
+      .catch((err) => {
+        setErrorMessage(err.message);
+        setIsLoading(false);
+        setError(false);
+      });
+  }, [single_topic]);
+
+  if (isLoading) {
+    return <p>Loading Topic....</p>;
```

Comment:
currently this loading message only appears when you first load the page

when switching between topics it does not re-appear!

this would not give a good user experience on slower devices